src/matrixvariate.jl

```diff
@@ -30,16 +30,20 @@ function TuringWishart(d::Wishart)
     return TuringWishart(d.df, getchol(d.S), d.logc0)
 end
 getchol(p::PDMat) = p.chol
-getchol(p::PDiagMat) = Diagonal(map(sqrt, p.diag))
-getchol(p::ScalMat) = Diagonal(fill(sqrt(p.value), p.dim))
+getchol(p::PDiagMat) = Cholesky(Diagonal(sqrt.(p.diag)), 'U', 0)
+getchol(p::ScalMat) = Cholesky(Diagonal(fill(sqrt(p.value), p.dim)), 'U', 0)
 
-function TuringWishart(df::T, S::AbstractMatrix) where {T <: Real}
-    p = size(S, 1)
-    df > p - 1 || error("dpf should be greater than dim - 1.")
+function TuringWishart(df::Real, S::AbstractMatrix{<:Real})
     C = cholesky(S)
     return TuringWishart(df, C)
 end
+function TuringWishart(df::Real, S::AbstractPDMat{<:Real})
+    C = getchol(S)
+    return TuringWishart(df, C)
+end
```

Comment:
I'd suggest to just define
```julia
getchol(x::AbstractMatrix{<:Real}) = cholesky(x)
TuringWishart(df::Real, S::AbstractMatrix{<:Real}) = TuringWishart(df, getchol(S))
```
and remove the existing definitions of `TuringWishart(df::Real, S::AbstractMatrix{<:Real})` and `TuringWishart(df::Real, S::AbstractPDMat{<:Real})` completely.

Reply:
Or maybe we could just use `cholesky` for all matrices? It is defined also for `AbstractPDMat`, so `getchol` could be removed completely and we just use
```julia
TuringWishart(df::Real, S::AbstractMatrix{<:Real}) = TuringWishart(df, cholesky(S))
```